src/data/finsupp/basic.lean

```diff
@@ -2192,6 +2200,16 @@ rfl
 
 end sum
 
+section
+
+variables [has_zero M] [monoid_with_zero R] [mul_action_with_zero R M]
+
+@[simp] lemma single_one_smul (a b : α) (f : α → M) :
+  (single a (1:R) b) • (f a) = single a (f b) b :=
+by { by_cases a = b; simp [h] }
```

Comment:
This feels like a bit of a weird lemma - I'll come back later to see if I can think of a more general statement.

Reply:
```lean
@[simp] lemma single_smul (a b : α) (f : α → M) (r : R) :
  (single a r b) • (f a) = single a (r • f b) b :=
by by_cases a = b; simp [h]
```

maybe this? I agree this is a bit weird, though, the only content is that case split